test-d/array-tail.ts

```diff
@@ -10,3 +10,17 @@ expectType<[]>(getArrayTail(['a', 'b', 'c']));
 expectType<[]>(getArrayTail([] as const));
 expectType<[]>(getArrayTail(['a'] as const));
 expectType<['b', 'c']>(getArrayTail(['a', 'b', 'c'] as const));
+
+// Optional elements tests
+expectType<[undefined, 'c']>(getArrayTail(['a', undefined, 'c'] as const));
+
+// Mixed optional/required
+type MixedArray = [string, undefined?, number?];
+expectType<[undefined?, number?]>(getArrayTail(['hello'] as MixedArray));
+
+// Optional numbers
+expectType<[undefined, 3]>(getArrayTail([1, undefined, 3] as const));
+
+// Complex mixed case
+type ComplexArray = [string, boolean, number?, string?];
+expectType<[boolean, number?, string?]>(getArrayTail(['test', false] as ComplexArray));
```

Comment:
@ferferga These test cases would have passed with the previous implementation as well, the problem was only when the first element was optional, like `["a"?, "b"?]`.

This test would have failed previously but works with updated implementation:
```ts
expectType<ArrayTail<['a'?, 'b'?]>>({} as ['b'?]);
```

<br>

Also, could add a test for unions:
```ts
expectType<ArrayTail<[] | ['a', 'b']>>({} as [] | ['b']);
```

Reply:
Added tests [here](https://github.com/sindresorhus/type-fest/pull/999).